addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CacheControlScanRule.java

```diff
@@ -65,11 +66,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
             String cacheControlHeaders =
                     (!cacheControlList.isEmpty()) ? cacheControlList.toString().toLowerCase() : "";
 
-            if (cacheControlHeaders.isEmpty()
+            if ((cacheControlHeaders.isEmpty()
                     || // No Cache-Control header at all
                     cacheControlHeaders.indexOf("no-store") < 0
                     || cacheControlHeaders.indexOf("no-cache") < 0
-                    || cacheControlHeaders.indexOf("must-revalidate") < 0) {
+                    || cacheControlHeaders.indexOf("must-revalidate") < 0)
+                    && // POST requests should not cache unless server asked to do so
+                    !HttpRequestHeader.POST.equals(msg.getRequestHeader().getMethod())) {
```

Comment:
Why not return earlier? No point doing all the checks if the rule is never going to raise the alert on POSTs.